### ISSUE TITLE
Option to auto-merge despite failing non-required CI checks

### ISSUE THREAD

**Issue by serhalp** [dependabot/dependabot-core#2661]
Hello!

I searched and couldn't find anything along these lines, other than #2132.

I'm not sure whether to classify this as a bug or a feature, but I guess [per the documentation](https://dependabot.com/docs/config-file/) this is expected behavior:
> For all of the options below Dependabot will wait until **all your status checks pass** before merging.

As mentioned by some folks in #2132, we use a number of different status checks on GitHub, some of which are required, because they are required, and some of which are not required, because they, well, are not required. I understand the conservative default behavior of not auto-merging PRs with a failing non-required check, but the inability to override this behavior is adding quite a bit of friction to our workflow.

In summary, when one or more non-required checks (as configured by GitHub branch protection rules) fail and all required checks pass, Dependabot comments with:

> One of your CI runs failed on this pull request, so Dependabot won't merge it.

and refuses to auto-merge the PR. My request is to allow toggling this behavior, at the organization level and/or at the repo level, to disregard failing non-required CI checks with respect to allowing auto-merge.

Thank you!

**Comment by elliotttf** [comment]
Sort of related to this, I have an optional check that's tied to an approval step in CircleCI. 

<img width="838" alt="Screen Shot 2021-11-23 at 13 23 14" src="https://user-images.githubusercontent.com/447151/143090313-8c46d576-2195-4a1b-924e-01b612d277dc.png">

Dependabot will not merge on approval (using `@dependabot squash and merge`) as long as this optional check is incomplete.

It seems it would be preferable for dependabot to merge once all required checks have passed (if any exist) or once all check have passed if none are required.

**Comment by jeffwidman** [comment]
👋 Hey there! Sorry for the glacially slow response... we've been a bit buried.

This issue is pretty old, from old Dependabot-preview before it was integrated with GitHub. With the new Dependabot native, we unfortunately are not going to be supporting native auto-merge for [the reasons outlined here](https://github.com/dependabot/dependabot-core/issues/1973#issuecomment-640918321). However, there is a [documented workaround if you wish to setup auto-merge yourself](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request).

I'm going to close, as essentially this issue isn't relevant anymore since the old way of doing automerge doesn't exist.

**Comment by bewuethr** [comment]
Sorry to dig up an old issue, but "using `@dependabot squash and merge` doesn't result in a merge" is a different issue from auto-merge, isn't it? I see it saying

> Dependabot will merge this PR once CI passes on it, as requested by @bewuethr.

and then never merge, even when all checks have passed.

Is there a better place to bring this up? This repo, or GitHub community discussions perhaps?
